src/Visualization.jl

```diff
@@ -28,27 +28,52 @@ function Makie.plot!(f::Makie.GridPosition, tn::TensorNetwork{A}; labels = false
     # TODO recognise them by using `DeltaArray` or `Diagonal` representations
     copytensors = findall(t -> haskey(t.meta, :dual), tensors(tn))
 
-    kwargs[:node_size] = [max(15, log2(size(tensors(tn, i)) |> prod)) for i in 1:nv(graph)]
-    kwargs[:node_marker] = [i ∈ copytensors ? :diamond : :circle for i in 1:length(tensors(tn))]
-    kwargs[:node_color] = [i ∈ copytensors ? :black : :white for i in 1:length(tensors(tn))]
+    oinds = openinds(tn)
+    ghostnodes = []
+    while !isempty(oinds)
+        for ind in oinds
+            tensor = links(ind)[1]
+            tensor_oinds = filter(id -> id.name ∈ tensor.labels, oinds)
```

Comment:
Use the getter methods to access the fields
```suggestion
            tensor_oinds = filter(id -> nameof(id) ∈ labels(tensor), oinds)
```